pennylane/transforms/core/transform_program.py

```diff
@@ -354,24 +352,32 @@ def set_classical_component(self, qnode, args, kwargs):
             self._set_all_classical_jacobians(qnode, args, kwargs, argnums)
             self._set_all_argnums(qnode, args, kwargs, argnums)
 
-    def prune_dynamic_transform(self):
-        """Ensure a single ``dynamic_one_shot`` transform is applied."""
-        trans_type = np.zeros(len(self._transform_program), dtype=np.int32)
-        for i, t in enumerate(self._transform_program):
-            if "dynamic_one_shot" in str(t):
-                trans_type[i] = 1
-            if "mid_circuit_measurements" in str(t):
-                trans_type[i] = 2
-        if sum(trans_type) < 2:
-            return
-        keep = 2 if 2 in trans_type else 1
+    def prune_dynamic_transform(self, type_to_keep=1):
+        """Ensures that only one or none ``dynamic_one_shot`` is applied.
```

Comment:
The name and first line don't seem to match what the method does. Would `check_dynamic_transform` be more appropriate?

Reply:
This method does prune the transform program, removing any duplicate `dynamic_one_shot` transforms. This is an existing public method, so I didn't want to change its name.

Reply:
and this name was set last release cycle. So I'm not sure we can change it at this point.

Reply:
>The name and first line don't seem to match what the method does.

Why is this? I seem to be overlooking the mismatch :thinking: